packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1299,13 +1299,86 @@ class RenderFractionalTranslation extends RenderProxyBox {
+  /// coordinate space configured such that the origin is at the top left of the
+  /// box. The area of the box is the size of the [size] argument.
+  ///
+  /// Paint operations should remain inside the given area. Graphical operations
+  /// outside the bounds may be silently ignored, clipped, or not clipped; no
+  /// guarantees are made.
+  ///
+  /// Implementations should be wary of correctly pairing any calls to
+  /// [Canvas.save]/[Canvas.saveLayer] and [Canvas.restore], otherwise all
+  /// subsequent painting on this canvas may be affected, with potentially
+  /// hilarious but confusing results.
+  ///
+  /// To paint text on a [Canvas], use a [TextPainter].
+  ///
+  /// To paint an image, obtain an [ImageResource] (e.g. using the
+  /// [ImageCache.load] method on the [imageCache] singleton), and create a new
+  /// instance of your custom delegate, giving it the [ImageResource] object's
+  /// [ImageInfo] object whenever it changes. Then, inside your delegate's
+  /// [paint] method, call the [Canvas.drawImage], [Canvas.drawImageRect], or
+  /// [Canvas.drawImageNine] methods to paint the [ImageInfo.image] object,
+  /// applying the [ImageInfo.scale] value to obtain the correct rendering size.
   void paint(Canvas canvas, Size size);
+
+  /// Called whenever a new instance of the custom painter delegate class is
+  /// provided to the [RenderCustomPaint] object, or any time that a new
+  /// [CustomPaint] object is created with a new instance of the custom painter
+  /// delegate class (which amounts to the same thing, since the latter is
+  /// implemented in terms of the former).
+  ///
+  /// If the new instance represents different information than the old
+  /// instance, then the method should return true, otherwise it should return
+  /// false.
+  ///
+  /// If the method returns false, then the paint call might be optimised away.
+  ///
+  /// It's possible that the [paint] method will get invoked even if
+  /// [shouldRepaint] returns false (e.g. if an ancestor or descendant needed to
+  /// be repainted). It's also possible that the [paint] method will get invoked
+  /// without [shouldRepaint] being called at all (e.g. if the box changes
+  /// size).
+  ///
+  /// If a custom delegate has a particularly expensive paint function such that
+  /// repaints should be avoided as much as possible, a [RepaintBoundary] or
+  /// [RenderRepaintBoundary] (or other render object with [isRepaintBoundary]
+  /// set to true) might be helpful.
   bool shouldRepaint(CustomPainter oldDelegate);
+
+  /// Called whenever a hit test is being performed on an object that is using
+  /// this custom paint delegate.
+  ///
+  /// The given point is relative to the same coordinate space as the last
+  /// [paint] call.
+  ///
+  /// The default behavior is to consider all points to not be hits for
+  /// foreground painters, and all points to be hits for background painters.
```

Comment:
I'd reverse this sentence because background paints are much more common than foreground painters.